src/pke/examples/simple-ckks-bootstrapping.cpp

```diff
@@ -0,0 +1,164 @@
+    * The SPARSE_TERNARY distribution was used in the original CKKS paper,
+    * but in this example, we use UNIFORM_TERNARY because this is included in the homomorphic
+    * encryption standard.
+    */
+    SecretKeyDist secretKeyDist = UNIFORM_TERNARY;
+    parameters.SetSecretKeyDist(secretKeyDist);
+
+    /*  A2) Desired security level based on FHE standards.
+    * In this example, we use the "NotSet" option, so the example can run more quickly with
+    * a smaller ring dimension. Note that this should be used only in
+    * non-production environments, or by experts who understand the security
+    * implications of their choices. In production-like environments, we recommend using
+    * HEStd_128_classic, HEStd_192_classic, or HEStd_256_classic for 128-bit, 192-bit,
+    * or 256-bit security, respectively. If you choose one of these as your security level,
+    * you do not need to set the ring dimension.
+    */
+    parameters.SetSecurityLevel(HEStd_NotSet);
+    parameters.SetRingDim(1 << 12);
+
+    /*  A3) Scaling parameters.
+    * By default, we set the modulus sizes and rescaling technique to the following values
+    * to obtain a good precision and performance tradeoff. We recommend keeping the parameters
+    * below unless you are an FHE expert.
+    */
+#if NATIVEINT == 128
+    ScalingTechnique rescaleTech = FIXEDAUTO;
+    usint dcrtBits               = 78;
+    usint firstMod               = 89;
+#else
+    ScalingTechnique rescaleTech = FLEXIBLEAUTO;
+    usint dcrtBits               = 59;
+    usint firstMod               = 60;
+#endif
+
+    parameters.SetScalingModSize(dcrtBits);
+    parameters.SetScalingTechnique(rescaleTech);
+    parameters.SetFirstModSize(firstMod);
+
+    /*  A4) Multiplicative depth.
+    * The goal of bootstrapping is to increase the number of available levels we have, or in other words,
+    * to dynamically increase the multiplicative depth. However, the bootstrapping procedure itself
+    * needs to consume a few levels to run. We compute the number of bootstrapping levels required
+    * using GetBootstrapDepth, and add it to levelsUsedBeforeBootstrap to set our initial multiplicative
+    * depth. We recommend using the input parameters below to get started.
+    */
+    std::vector<uint32_t> levelBudget = {4, 4};
+    uint32_t approxBootstrapDepth     = 9;
+
+    uint32_t levelsUsedBeforeBootstrap = 30;
+    usint depth =
+        levelsUsedBeforeBootstrap + FHECKKSRNS::GetBootstrapDepth(approxBootstrapDepth, levelBudget, secretKeyDist);
+    parameters.SetMultiplicativeDepth(depth);
+
+    CryptoContext<DCRTPoly> cryptoContext = GenCryptoContext(parameters);
+
+    cryptoContext->Enable(PKE);
+    cryptoContext->Enable(KEYSWITCH);
+    cryptoContext->Enable(LEVELEDSHE);
+    cryptoContext->Enable(ADVANCEDSHE);
+    cryptoContext->Enable(FHE);
+
+    usint ringDim  = cryptoContext->GetRingDimension();
+    usint numSlots = ringDim / 2;
```

Comment:
Add a comment that this is the maximum number of slots that can be used for full packing.